docs/advanced/experimental/compound_executors.md

```diff
@@ -0,0 +1,52 @@
+(compound-executor)=
+# Compound Executors
+
+A Compound Executor is an advanced type of Executor. 
+This combines two or more Executors into one class, in order to facilitate special logic.
+Examples of this include:
+
+- combining a Storage and a Search Indexer in one, for easier access, scaling etc.
+- combining a custom ranker and an Indexer in one, for cases where `top_k` is not achieved after a ranking process filters out results
+- having Indexers that store multiple level of granularity in one Executor, for easier lookup and custom logic
+
+If you want to develop a `CompoundExecutor`-type Executor based on one of the Executors existing in Jina Hub, you have to copy-paste the code of the classes you need into your own Executor's package.
+This is because, for now, the Executors in Jina Hub can **not** be imported as Python modules.
+
+## Example
```

Comment:
Should this be inside some nice ```admonition thingy for better style? Not sure how other pages do it.

Reply:
Nevermind, only the code should go into an example as in https://docs.jina.ai/fundamentals/concepts/, or?

Reply:
You mean the code?

It looks good to me 

![Screenshot from 2021-09-15 12-18-38](https://user-images.githubusercontent.com/8330330/133416099-c3d8fdf2-16a2-4f9a-ba7d-a16da53ce740.png)

Reply:
Ah I see what you mean. I think it looks better without the admonition. The admonition makes it looks smaller and harder to read.

Reply:
![Screenshot from 2021-09-15 12-45-28](https://user-images.githubusercontent.com/8330330/133419876-17b2e518-c090-42de-b6e0-ef3dba6421f3.png)

Reply:
While I see what you mean, I think the general style should be consistent. If such examples belong in an admonition, this should go there as well. But I am not sure, if it should go into one, so keep it as it is. @alaeddine-13 perhaps you can comment on it, since you transferred quite some cookbooks into docs.

Reply:
But they're not all like that 

ex http://localhost:63342/executors/docs/_build/dirhtml/fundamentals/executor/executor-api/ http://localhost:63342/executors/docs/_build/dirhtml/fundamentals/flow/async-flow/